code/game/objects/items/stacks/tiles/tile_reskinning.dm

```diff
@@ -0,0 +1,70 @@
+/////////////////////
+// Tile reskinning //
+/////////////////////
+// Q: What is this?
+// A: A simple function to allow you to change what tiles you place with a stack of tiles.
+// Q: Why do it this way?
+// A: This allows players more freedom to do beautiful-looking builds. Having five types of titanium tile would be clunky as heck.
+// Q: Great! Can I use this for all floors?
+// A: Unfortunately, this does not work on subtypes of plasteel and instead we must change the icon_state of these turfs instead, as the icon_regular_floor var that "saves" what type of floor a plasteel subtype turf was so once repaired...
+// ... it'll go back to the floor it was instead of grey (medical floors turn white even after crowbaring the tile and putting it back). This stops changing turf_type from working.
+
+GLOBAL_LIST_EMPTY(titanium_tile_reskinning_radial_choices)
+
+/obj/item/stack/tile/mineral/titanium/attack_self(mob/user)
+	var/list/choices = GLOB.titanium_tile_reskinning_radial_choices
+	if(!length(choices))
+		choices = GLOB.titanium_tile_reskinning_radial_choices = list(
```

Comment:
```suggestion
/obj/item/stack/tile/mineral/titanium/attack_self(mob/user)
	var/static/list/choices = list(
```
Since this won't really be used by anything else you can just make it a proc-exclusive static list.
For reference, this will be only be populated the first time the proc is called, and re-used on further proc calls.